c++/src/kj/compat/http.c++

```diff
@@ -3677,6 +4095,41 @@ public:
               });
             }
 
+            kj::Maybe<CompressionParameters> compressionParameters;
+            if (settings.enableWebSocketCompression) {
+              KJ_IF_MAYBE(agreedParameters, responseHeaders.get(
+                  HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
+                auto parseResult = _::tryParseExtensionAgreement(offeredExtensions,
+                    *agreedParameters);
+                if (parseResult.is<kj::Exception>()) {
+                  return settings.errorHandler.orDefault(*this).handleWebSocketProtocolError({
+                    502, "Bad Gateway", parseResult.get<kj::Exception>().getDescription(), nullptr});
+                }
+
+                compressionParameters = kj::mv(parseResult.get<CompressionParameters>());
+                auto& agreement = KJ_ASSERT_NONNULL(compressionParameters);
+                KJ_IF_MAYBE(original, clientOffer) {
+                  // Client offered `client_x` parameters, let's compare them with the response.
```

Comment:
Nit: this comment should probably be updated. Due to changes in how we parse the clients request (before sending it to the server), `clientOffer` will only be `null` if the client _does not_ send any `permessage-deflate` extensions. That is, we enter this `KJ_IF_MAYBE` even if the client offers `permessage-deflate` without any `client_x` parameters.

Also, changing `KJ_IF_MAYBE` to `const auto& original = KJ_ASSERT_NONNULL(clientOffer)` would make the intention more clear, since we always expect `clientOffer` to be `!null` at this point (we throw in `tryParseExtensionAgreement` if `offeredExtensions` is `null`).

Actually, we don't even need to pass `offeredExtensions` to this `.then()` now that `clientOffer` has changed in this way. We could just pass `clientOffer` to `tryParseExtensionAgreement` (instead of `offeredExtensions`) and move the stuff in the block starting at line 4111 into `tryParseExtensionAgreement`.

Reply:
Basically this: https://github.com/capnproto/capnproto/pull/1480/commits/03b65e6f59754f7faa1041fdcf0d629aa4ccb7f3, if this is alright I'll rebase and we can merge, otherwise I'll revert the fixup.